lib/fabrik/DirectorManager.js

```diff
@@ -211,6 +211,31 @@ class DirectorManager extends BaseManager {
                 previous_manifest: manifest
               }, opts.context);
               return;
+            })
+            .then(() => {
+              let updateActionContext = _.cloneDeep(actionContext);
+              _.chain(updateActionContext)
+                .set('instance_guid', this.getInstanceGuid(deploymentName))
+                .value();
+              _.chain(updateActionContext.params)
+                .set('service_id', this.service.id)
+                .set('plan_id', this.plan.id)
+                .value();
+              return this
+                .getDeploymentIps(deploymentName)
+                .then(ips => this.agent.preUpdate(ips, updateActionContext))
+                .tap(response => preUpdateResponse = response.actionResponse)
+                .catch(FeatureNotSupportedByAnyAgent, ServiceInstanceNotOperational, err => {
+                  logger.debug('+-> Catched expected error of feature \'preUpdate\':', err);
+                  return;
+                })
+                .catch(err => {
+                  if (err.status === CONST.HTTP_STATUS_CODE.NOT_FOUND) {
+                    logger.debug('+-> Catched expected error of feature \'preUpdate\':', err);
+                    return;
+                  }
+                  throw err;
```

Comment:
can you log the error here before throwing back ? Or will the calling layer log it ?

Reply:
Will get logged here: https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/246/files#diff-52f07cb68d46ec974943b67d2c207a47R262

Reply:
ok ..